sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -172,6 +182,28 @@ private boolean shouldApplyScopeData(
     return sentryId;
   }
 
+  private boolean shouldSendSessionUpdate(
```

Comment:
I've read https://github.com/getsentry/sentry-java/pull/2001#discussion_r855826594 but I'm not sure about it.
Why don't we want to send sessions if the state has not changed? is that because the session will be exactly the same? if so, not only the state changes but also the `errorCount` and `userAgent` as well.

Reply:
I'd rather prefer the `SentryClient#updateSessionData` returning a pair of the `Session` and a boolean which means: the session got updated, so you don't need this `shouldSendSessionUpdate` logic.

Reply:
Hmm it's not just about any update to the session. For this we only care about some specific updates (session health) that might have happened. Doing this as part of the update method doesn't feel quite right to me.

Reply:
The method should be called `shouldSendSessionUpdateForDroppedEvent` to better reflect what the check is for.

Reply:
Renaming the method is fine, I'd just prefer that the method that actually updates the session returns a boolean telling it's updated or not, instead of this logic that actually checks only a few things.
I'm not quite sure this logic makes sense actually, maybe we'd need to jump on a call to get your explanation since I was not part of the whole discussion about this bug specifically.